### PR TITLE
feat: add custom darglayer

### PR DESCRIPTION
**Pull request by DennyWanye**
add custom darglayer to make the drag effect manageable.

- [x] add custom darglayer
- [x] hide original darg target
- [x] optimize the UI

### CODE REVIEW

**Comment by DennyWanye** [comment]
for the thrid task, i didn't find a better way to change the interaction effect. So I will close the thrid task and leave a todo.

File: packages/ui-components/src/Tabs/CustomDragLayer.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { DragLayer } from 'react-dnd';
+import { Types } from './TabHeader';
+const layerStyles = {
```
Comment (iNorthIsle) [inline packages/ui-components/src/Tabs/CustomDragLayer.js:4]: please use class instead, btw, need to format before u commit

File: packages/ui-components/src/Tabs/TabHeader.js [hunk 2]
```diff
@@ -123,6 +125,15 @@ class TabHeaderItem extends PureComponent {
     )
   }
 
+  componentDidMount() {
+    const { connectDragPreview } = this.props;
+    if (connectDragPreview) {
```
Comment (iNorthIsle) [inline packages/ui-components/src/Tabs/TabHeader.js:130]:
```suggestion
    connectDragPreview && connectDragPreview(getEmptyImage(), {
      captureDraggingState: true,
    });
```

File: packages/ui-components/src/Tabs/CustomDragLayer.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+};
+function getItemStyles (props) {
+    const { initialOffset, currentOffset } = props
+    if (!initialOffset || !currentOffset) {
+        return {
+            display: 'none'
+        }
+    }
+
+    let { x, y } = currentOffset
```
Comment (iNorthIsle) [inline packages/ui-components/src/Tabs/CustomDragLayer.js:21]: const { x } = currentOffset
    const transform = `translate(${x}px, ${initialOffset.y}px)`
    return {
        transform,
        WebkitTransform: transform
    }

**Pull request closed without merging**